sql/core/src/test/scala/org/apache/spark/sql/execution/SparkSqlSerializer2Suite.scala

```diff
@@ -57,6 +56,8 @@ class SparkSqlSerializer2DataTypeSuite extends SparkFunSuite {
   checkSupported(DecimalType(10, 5), isSupported = true)
   checkSupported(DecimalType.Unlimited, isSupported = true)
 
+  // If NullType is the only data type in the schema, we do not support it.
```

Comment:
For completeness, do we need a test to check that schemas that contain a NullType column and a column of some other supported type are supported?